libs/barista-components/filter-field/src/filter-field-multi-select/filter-field-multi-select.ts

```diff
@@ -0,0 +1,344 @@
+      this._template,
+      this._viewContainerRef,
+    );
+
+    // init keymanager with options
+    this._keyManager = new ActiveDescendantKeyManager<DtOption<T>>(
+      this._options,
+    ).withWrap();
+
+    this._options.changes
+      .pipe(
+        startWith(null),
+        switchMap(() =>
+          merge(...this._options.map((option) => option._optionHovered)),
+        ),
+        takeUntil(this._destroy$),
+      )
+      .subscribe((option) => {
+        this._ngZone?.run(() => {
+          this._keyManager.setActiveItem(option);
+        });
+      });
+  }
+
+  /**
+   * @internal
+   * Sets the panel scrollTop. This allows us to manually scroll to display options
+   * above or below the fold, as they are not actually being focused when active.
+   */
+  _setScrollTop(scrollTop: number): void {
+    if (this._panel) {
+      this._panel.nativeElement.scrollTop = scrollTop;
+    }
+  }
+
+  /**
+   * @internal
+   * Returns the panel's scrollTop.
+   */
+  _getScrollTop(): number {
+    return this._panel ? this._panel.nativeElement.scrollTop : 0;
+  }
+
+  /** @internal Marks the filter-field multiSelect for change detection. */
+  _markForCheck(): void {
+    this._changeDetectorRef.markForCheck();
+  }
+
+  /** @internal Unique ID to be used on a local element */
+  _getLocalId(suffix: string): string {
+    return `${this.id}-${suffix}`;
+  }
+
+  /**  Handles the submit of multiSelect values. */
+  handleSubmit(event: Event): void {
+    event.preventDefault();
+    event.stopImmediatePropagation();
+
+    this._emitSelectEvent();
+  }
+
+  /**
+   * @internal
+   * Emits the `select` event.
+   */
+  _emitSelectEvent(): void {
+    this.multiSelectSubmitted.emit(
+      new DtFilterFieldMultiSelectSubmittedEvent(this, this._currentSelection),
+    );
+    // After emission we need to reset the multiSelect state, to have a fresh one
+    // if another multiSelect opens.
+    this._currentSelection = [];
+  }
+
+  /** @internal Set pre selected options for the multiSelect input fields. */
+  _setInitialSelection(values: T[]): void {
+    if (Array.isArray(values)) {
+      this._initialSelection = values;
+      this._currentSelection = values.slice();
+    } else {
+      this._initialSelection = [];
+    }
+    this._checkApplyDisable();
+  }
+
+  /** Toggle option */
+  toggleOption(option: Highlightable & DtOption<T>): void {
+    this._currentSelection.includes(option.value)
+      ? (this._currentSelection = this._currentSelection.filter(
+          (opt) => opt != option.value,
+        ))
+      : this._currentSelection.push(option.value);
+    this._checkApplyDisable();
+  }
+
+  /** @internal Toggle option from template */
+  _toggleOptionFromTemplate(event: DtCheckboxChange<T>): void {
+    if (event.checked) this._currentSelection.push(event.source.value);
+    else
+      this._currentSelection = this._currentSelection.filter(
+        (opt) => opt !== event.source.value,
+      );
```

Comment:
```suggestion
    if (event.checked) {
      this._currentSelection.push(event.source.value);
    } else {
      this._currentSelection = this._currentSelection.filter(
        (opt) => opt !== event.source.value,
      );
    }
```
Please use curly brackets for if statements - this is a potential pitfall